lib/src/objdetect/objdetect_async.dart

```diff
@@ -0,0 +1,917 @@
+library cv;
+
+import 'dart:ffi' as ffi;
+import 'dart:typed_data';
+import 'package:ffi/ffi.dart';
+import '../core/base.dart';
+import '../core/mat.dart';
+import '../core/point.dart';
+import '../core/rect.dart';
+import '../core/size.dart';
+import '../core/vec.dart';
+import '../opencv.g.dart' as cvg;
+
+class CascadeClassifier extends CvStruct<cvg.CascadeClassifier> {
+  CascadeClassifier._(cvg.CascadeClassifierPtr ptr, [bool attach = true])
+      : super.fromPointer(ptr) {
+    if (attach) {
+      finalizer.attach(this, ptr.cast(), detach: this);
+    }
+  }
+
+  factory CascadeClassifier.empty() {
+    final p = calloc<cvg.CascadeClassifier>();
+    cvRun(() => CFFI.CascadeClassifier_New(p));
+    return CascadeClassifier._(p);
+  }
+
+  factory CascadeClassifier.fromFile(String filename) {
+    final p = calloc<cvg.CascadeClassifier>();
+    final cp = filename.toNativeUtf8().cast<ffi.Char>();
+    cvRun(() => CFFI.CascadeClassifier_NewFromFile(cp, p));
+    calloc.free(cp);
+    return CascadeClassifier._(p);
+  }
```

Comment:
Do you think that we need to async constructors? Loading the model is a heavy operation, if users load it in main thread, especially in Flutter, UI may be blocked.

Reply:
Yes I think so too, will check if it's doable and let you know

Reply:
Yes, use static methods.

> they are identical so that is not a problem i think, also async constructors will be named async right?

Yes, use `static Future<CascadeClassifier> emptyAsync() async {}` will be clearer.

to split sync and async to different files, maybe some thing like https://github.com/rainyl/opencv_dart/issues/118#issuecomment-2194193568 will be better?

Reply:
I like that, will do it like the comment it clear and separated, so will do and the async constructors